{
  "nbformat": 4,
  "nbformat_minor": 0,
  "metadata": {
    "colab": {
      "provenance": [],
      "include_colab_link": true
    },
    "kernelspec": {
      "name": "python3",
      "display_name": "Python 3"
    },
    "language_info": {
      "name": "python"
    }
  },
  "cells": [
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "view-in-github",
        "colab_type": "text"
      },
      "source": [
        "<a href=\"https://colab.research.google.com/github/joscarhuguet/Genome-Data-Mining-and-Pathogen-Identification/blob/main/fastq_QC_evaluation.ipynb\" target=\"_parent\"><img src=\"https://colab.research.google.com/assets/colab-badge.svg\" alt=\"Open In Colab\"/></a>"
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "# Fastq Quality Evaluation"
      ],
      "metadata": {
        "id": "HrUYhNfEZ6yM"
      }
    },
    {
      "cell_type": "markdown",
      "source": [
        "This notebook will evaluate the quality of FASTQ files generated by sequencing technologies: Illumina short reads and Nanopore long reads."
      ],
      "metadata": {
        "id": "ekHOVr-vYnaP"
      }
    },
    {
      "cell_type": "markdown",
      "source": [
        "First, we will install the required dependencies using conda, including FastQC and NanoPlot."
      ],
      "metadata": {
        "id": "IISHN8i5ZUYd"
      }
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "48cfda8e"
      },
      "source": [
        "!apt-get update\n",
        "!apt-get install -y fastqc"
      ],
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "!pip install nanoplot -q\n",
        "!apt-get install -y libfreetype6 libpng-dev"
      ],
      "metadata": {
        "id": "lzso2fQsXcCM"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [
        "Next, we will load the Illumina short-read data into our workspace using the wget command. We will run FastQC on these reads and evaluate the results."
      ],
      "metadata": {
        "id": "yPAf-3MxZbp7"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "!wget http://ftp.sra.ebi.ac.uk/vol1/fastq/SRR240/022/SRR24008622/SRR24008622_1.fastq.gz\thttp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR240/022/SRR24008622/SRR24008622_2.fastq.gz\n"
      ],
      "metadata": {
        "id": "rGRZElcOS83A"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "!fastqc *.fastq.gz"
      ],
      "metadata": {
        "id": "KYyZ6txvTDnG"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [
        "The cell below will help us to visualize the results"
      ],
      "metadata": {
        "id": "ZG2eUASiZw9Y"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "import os\n",
        "from IPython.core.display import display, HTML\n",
        "\n",
        "# Ask the user for the file name they want to display\n",
        "file_name = input(\"Enter the name of the HTML file you want to display (include .html extension): \")\n",
        "\n",
        "# Check if the file exists\n",
        "if os.path.exists(file_name):\n",
        "    # Open and read the HTML file\n",
        "    with open(file_name, 'r') as file:\n",
        "        html_content = file.read()\n",
        "        display(HTML(html_content))  # Display the HTML content\n",
        "else:\n",
        "    print(f\"File '{file_name}' not found. Please ensure the file exists in the current directory.\")\n"
      ],
      "metadata": {
        "id": "MGDM7kvdWmRg"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [
        " Now we will fetch the Nanopore long-read data using wget, run NanoPlot, and evaluate the output."
      ],
      "metadata": {
        "id": "T8RwV8KuZiyx"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "!wget https://zenodo.org/record/14018699/files/SRR30576370.fastq.gz"
      ],
      "metadata": {
        "id": "fNNx1ADZX0KV"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "!NanoPlot --fastq SRR30576370.fastq.gz"
      ],
      "metadata": {
        "id": "8J7FWnjjWvYa"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [
        "The cell below will help us to visualize the results"
      ],
      "metadata": {
        "id": "GIDDVQLlZ4ea"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "import os\n",
        "from IPython.core.display import display, HTML\n",
        "\n",
        "# Ask the user for the file name they want to display\n",
        "file_name = input(\"Enter the name of the HTML file you want to display (include .html extension): \")\n",
        "\n",
        "# Check if the file exists\n",
        "if os.path.exists(file_name):\n",
        "    # Open and read the HTML file\n",
        "    with open(file_name, 'r') as file:\n",
        "        html_content = file.read()\n",
        "        display(HTML(html_content))  # Display the HTML content\n",
        "else:\n",
        "    print(f\"File '{file_name}' not found. Please ensure the file exists in the current directory.\")\n"
      ],
      "metadata": {
        "id": "VzXg7mtUYGrP"
      },
      "execution_count": null,
      "outputs": []
    }
  ]
}