{
  "nbformat": 4,
  "nbformat_minor": 0,
  "metadata": {
    "colab": {
      "provenance": []
    },
    "kernelspec": {
      "name": "python3",
      "display_name": "Python 3"
    },
    "language_info": {
      "name": "python"
    }
  },
  "cells": [
    {
      "cell_type": "markdown",
      "source": [
        "# Fastq Quality Evaluation"
      ],
      "metadata": {
        "id": "HrUYhNfEZ6yM"
      }
    },
    {
      "cell_type": "markdown",
      "source": [
        "This notebook will evaluate the quality of FASTQ files generated by sequencing technologies: Illumina short reads and Nanopore long reads."
      ],
      "metadata": {
        "id": "ekHOVr-vYnaP"
      }
    },
    {
      "cell_type": "markdown",
      "source": [
        "First, we will install the required dependencies using conda, including FastQC and NanoPlot."
      ],
      "metadata": {
        "id": "IISHN8i5ZUYd"
      }
    },
    {
      "cell_type": "code",
      "metadata": {
        "id": "48cfda8e"
      },
      "source": [
        "!apt-get update\n",
        "!apt-get install -y fastqc"
      ],
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "!pip install nanoplot -q\n",
        "!apt-get install -y libfreetype6 libpng-dev"
      ],
      "metadata": {
        "id": "lzso2fQsXcCM",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "aee8e488-a54c-4a2f-fdc8-301a47ddccb8"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "  Preparing metadata (setup.py) ... \u001b[?25l\u001b[?25hdone\n",
            "  Preparing metadata (setup.py) ... \u001b[?25l\u001b[?25hdone\n",
            "\u001b[33m  DEPRECATION: Building 'nanoplot' using the legacy setup.py bdist_wheel mechanism, which will be removed in a future version. pip 25.3 will enforce this behaviour change. A possible replacement is to use the standardized build interface by setting the `--use-pep517` option, (possibly combined with `--no-build-isolation`), or adding a `pyproject.toml` file to the source tree of 'nanoplot'. Discussion can be found at https://github.com/pypa/pip/issues/6334\u001b[0m\u001b[33m\n",
            "\u001b[0m  Building wheel for nanoplot (setup.py) ... \u001b[?25l\u001b[?25hdone\n",
            "\u001b[33m  DEPRECATION: Building 'nanoget' using the legacy setup.py bdist_wheel mechanism, which will be removed in a future version. pip 25.3 will enforce this behaviour change. A possible replacement is to use the standardized build interface by setting the `--use-pep517` option, (possibly combined with `--no-build-isolation`), or adding a `pyproject.toml` file to the source tree of 'nanoget'. Discussion can be found at https://github.com/pypa/pip/issues/6334\u001b[0m\u001b[33m\n",
            "\u001b[0m  Building wheel for nanoget (setup.py) ... \u001b[?25l\u001b[?25hdone\n",
            "Reading package lists... Done\n",
            "Building dependency tree... Done\n",
            "Reading state information... Done\n",
            "libpng-dev is already the newest version (1.6.37-3build5).\n",
            "libfreetype6 is already the newest version (2.11.1+dfsg-1ubuntu0.3).\n",
            "libfreetype6 set to manually installed.\n",
            "0 upgraded, 0 newly installed, 0 to remove and 41 not upgraded.\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "Next, we will load the Illumina short-read data into our workspace using the wget command. We will run FastQC on these reads and evaluate the results."
      ],
      "metadata": {
        "id": "yPAf-3MxZbp7"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "!wget http://ftp.sra.ebi.ac.uk/vol1/fastq/SRR240/022/SRR24008622/SRR24008622_1.fastq.gz\thttp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR240/022/SRR24008622/SRR24008622_2.fastq.gz\n"
      ],
      "metadata": {
        "id": "rGRZElcOS83A"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "!fastqc *.fastq.gz"
      ],
      "metadata": {
        "id": "KYyZ6txvTDnG"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [
        "The cell below will help us to visualize the results"
      ],
      "metadata": {
        "id": "ZG2eUASiZw9Y"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "import os\n",
        "from IPython.core.display import display, HTML\n",
        "\n",
        "# Ask the user for the file name they want to display\n",
        "file_name = input(\"Enter the name of the HTML file you want to display (include .html extension): \")\n",
        "\n",
        "# Check if the file exists\n",
        "if os.path.exists(file_name):\n",
        "    # Open and read the HTML file\n",
        "    with open(file_name, 'r') as file:\n",
        "        html_content = file.read()\n",
        "        display(HTML(html_content))  # Display the HTML content\n",
        "else:\n",
        "    print(f\"File '{file_name}' not found. Please ensure the file exists in the current directory.\")\n"
      ],
      "metadata": {
        "id": "MGDM7kvdWmRg"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [
        " Now we will fetch the Nanopore long-read data using wget, run NanoPlot, and evaluate the output."
      ],
      "metadata": {
        "id": "T8RwV8KuZiyx"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "!wget https://zenodo.org/record/14018699/files/SRR30576370.fastq.gz"
      ],
      "metadata": {
        "id": "fNNx1ADZX0KV"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "!NanoPlot --fastq SRR30576370.fastq.gz"
      ],
      "metadata": {
        "id": "8J7FWnjjWvYa"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [
        "The cell below will help us to visualize the results"
      ],
      "metadata": {
        "id": "GIDDVQLlZ4ea"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "import os\n",
        "from IPython.core.display import display, HTML\n",
        "\n",
        "# Ask the user for the file name they want to display\n",
        "file_name = input(\"Enter the name of the HTML file you want to display (include .html extension): \")\n",
        "\n",
        "# Check if the file exists\n",
        "if os.path.exists(file_name):\n",
        "    # Open and read the HTML file\n",
        "    with open(file_name, 'r') as file:\n",
        "        html_content = file.read()\n",
        "        display(HTML(html_content))  # Display the HTML content\n",
        "else:\n",
        "    print(f\"File '{file_name}' not found. Please ensure the file exists in the current directory.\")\n"
      ],
      "metadata": {
        "id": "VzXg7mtUYGrP"
      },
      "execution_count": null,
      "outputs": []
    }
  ]
}